### PR TITLE
Add methods to integrate secret population steps

### PR DESCRIPTION
**Pull request by protatremy**
creation and deletion

* [ ] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/test/integration/test_integration_with_secrets.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    c['schedulers'] = [
+        schedulers.ForceScheduler(
+            name="force",
+            builderNames=["testy"])]
+
+    c['secretsProviders'] = [FakeSecretStorage(
+        secretdict={"foo": "bar", "something": "more"})]
+    f = BuildFactory()
+    f.addSteps([steps.ShellCommand(command=Interpolate('echo %(secret:foo)s'))],
```
Comment (tardyp) [inline master/buildbot/test/integration/test_integration_with_secrets.py]: need another test with the form
```
f.withSecrets([..]):
   f.addStep(...)
```
You can parametrize this function
```
def masterConfig(use_with_form=False):
    ...
```

File: master/buildbot/test/integration/test_integration_with_secrets.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+               withSecret=[("pathA", Interpolate('%(secret:something)s'))])
+    c['builders'] = [
+        BuilderConfig(name="testy",
+                      workernames=["local1"],
+                      factory=f)]
+    return c
+
+
+# master configuration with context manager function
+def masterConfigWith():
```
Comment (tardyp) [inline master/buildbot/test/integration/test_integration_with_secrets.py]: please factorize the two masterConfig into one function

File: master/docs/manual/secretsmanagement.rst [hunk 3]
```diff
@@ -94,6 +94,33 @@ In the master configuration, the Vault provider is instantiated through the Buil
...
+How to populate secrets in a build
+----------------------------------
+
+To populate secrets in files during a build, 2 steps are used to create and delete the files on the worker.
+The files will be automatically deleted at the end of the build.
+
+.. code-block:: python
+
+        f = BuildFactory()
+        with f.withSecret(secrets_list):
```
Comment (tardyp) [inline master/docs/manual/secretsmanagement.rst]: as it is a list, it should be ```withSecrets```

File: master/docs/manual/secretsmanagement.rst [hunk 4]
```diff
@@ -94,6 +94,33 @@ In the master configuration, the Vault provider is instantiated through the Buil
...
+            f.addStep(step_definition)
+ or
+
+.. code-block:: python
+
+        f = BuildFactory()
+        f.addSteps([list_of_step_definitions], withSecret=[secrets_list])
```
Comment (tardyp) [inline master/docs/manual/secretsmanagement.rst]: withSecrets

File: master/buildbot/steps/download_secret_to_worker.py [hunk 5]
```diff
@@ -25,30 +23,22 @@
...
     def runPopulateSecrets(self):
         all_results = []
-        for secret in self.secret_to_be_populated:
-            res = yield self.downloadFileContentToWorker(secret.path, secret.secretvalue)
+        for path, secretvalue in self.secret_to_be_populated:
+            if not isinstance(path, str):
+                raise ValueError("Secret path %s is not a string" % path)
+            self.secret_to_be_interpolated = secretvalue
+            res = yield self.downloadFileContentToWorker(path, self.secret_to_be_interpolated)
             all_results.append(res)
```
Comment (tardyp) [inline master/buildbot/steps/download_secret_to_worker.py]: You should rather use buildbot.results.worstStatus

File: master/buildbot/test/integration/test_integration_with_secrets.py (hunk 2 above)
Comment (protatremy) [inline master/buildbot/test/integration/test_integration_with_secrets.py]: done

File: master/docs/manual/secretsmanagement.rst (hunk 3 above)
Comment (protatremy) [inline master/docs/manual/secretsmanagement.rst]: done

File: master/docs/manual/secretsmanagement.rst (hunk 4 above)
Comment (protatremy) [inline master/docs/manual/secretsmanagement.rst]: done

File: master/buildbot/steps/download_secret_to_worker.py (hunk 5 above)
Comment (protatremy) [inline master/buildbot/steps/download_secret_to_worker.py]: done

**Pull request merged**